src/test/java/com/uid2/admin/vertx/ClientSideKeypairServiceTest.java

```diff
@@ -481,7 +511,34 @@ void updateKeypairDisabledAndContact(Vertx vertx, VertxTestContext testContext)
 
         post(vertx, "api/client_side_keypairs/update", jo.encode(), testContext.succeeding(response -> testContext.verify(() -> {
             assertEquals(200, response.statusCode());
-            ClientSideKeypair expected = new ClientSideKeypair("89aZ234567", pub1, priv1, 124, "updated@email.com", time, false);
+            ClientSideKeypair expected = new ClientSideKeypair("89aZ234567", pub1, priv1, 124, "updated@email.com", time, false, name1);
+            validateKeypair(expected, response.bodyAsJsonObject());
+            testContext.completeNow();
+        })));
+    }
+
+    @Test
+    void updateKeypairDisabledAndName(Vertx vertx, VertxTestContext testContext) throws Exception {
+        fakeAuth(Role.ADMINISTRATOR);
+
+        Instant time = Instant.now();
+        Map<String, ClientSideKeypair> expectedKeypairs = new HashMap<>() {{
+            put("89aZ234567", new ClientSideKeypair("89aZ234567", pub1, priv1, 124, "test-two@example.com", time, true, name1));
+            put("9aZ2345678", new ClientSideKeypair("9aZ2345678", pub2, priv2, 123, "test@example.com", Instant.now(), true, name2));
+            put("789aZ23456", new ClientSideKeypair("789aZ23456", pub3, priv3, 125, "test-two@example.com", Instant.now(), false, name3));
+        }};
+
+        setKeypairs(new ArrayList<>(expectedKeypairs.values()));
+        setSites(new Site(123, "test", true));
+
+        JsonObject jo = new JsonObject();
+        jo.put("subscription_id", "89aZ234567");
+        jo.put("name", "updated name");
+        jo.put("disabled", false);
+
+        post(vertx, "api/client_side_keypairs/update", jo.encode(), testContext.succeeding(response -> testContext.verify(() -> {
+            assertEquals(200, response.statusCode());
+            ClientSideKeypair expected = new ClientSideKeypair("89aZ234567", pub1, priv1, 124, "test-two@example.com", time, false, "updated name");
             validateKeypair(expected, response.bodyAsJsonObject());
```

Comment:
Could you add check like this `verify(keypairStoreWriter, times(1)).upload(any(), isNull());` to tests to make sure upload is/isn't called.

Reply:
I added this check to all tests that add or update keypairs.